anomalib/models/rkde/region_extractor.py

```diff
@@ -0,0 +1,139 @@
+"""Region-based Anomaly Detection with Real Time Training and Analysis.
+
+Region Extractor.
+"""
+
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from enum import Enum
+from typing import List
+
+import torch
+from torch import Tensor, nn
+from torchvision.models.detection import fasterrcnn_resnet50_fpn
+from torchvision.ops import boxes as box_ops
+
+from anomalib.data.utils.boxes import scale_boxes
+
+
+class RoiStage(str, Enum):
+    """Processing stage from which rois are extracted."""
+
+    RCNN = "rcnn"
+    RPN = "rpn"
+
```

Comment:
Should we add
```python
def __str__(self):
    return self.value

Reply:
Not sure if this is needed tbh. This also works fine without adding `__str__`:
```python
>>> RoiStage.RCNN == "rcnn"
True
>>> self.stage
<RoiStage.RCNN: 'rcnn'>
>>> self.stage == RoiStage.RCNN
True
```

Reply:
Ah got it!